src/gui/application.cpp

```diff
@@ -141,18 +141,32 @@ bool Application::configVersionMigration()
 
     // Did the client version change?
     // (The client version is adjusted further down)
-    bool versionChanged = configFile.clientVersionString() != MIRALL_VERSION_STRING;
+    const auto versionChanged = configFile.clientVersionString() != MIRALL_VERSION_STRING;
 
     // We want to message the user either for destructive changes,
     // or if we're ignoring something and the client version changed.
-    bool warningMessage = !deleteKeys.isEmpty() || (!ignoreKeys.isEmpty() && versionChanged);
+    const auto warningMessage = !deleteKeys.isEmpty() || (!ignoreKeys.isEmpty() && versionChanged);
 
-    if (!versionChanged && !warningMessage)
+    if (!versionChanged && !warningMessage) {
         return true;
+    }
 
-    const auto backupFile = configFile.backup();
+    // back up all old config file
+    QStringList backupFilesList;
+    QDir configDir(configFile.configPath());
+    const auto anyConfigFileList = configDir.entryInfoList({"*.cfg"}, QDir::Files);
+    for (const auto &oldConfig : anyConfigFileList) {
+        const auto oldConfigFile = oldConfig.fileName();
+        const auto newConfigFile = configFile.configFile();
+        backupFilesList.append(configFile.backup(oldConfigFile));
+        if (oldConfigFile != newConfigFile) {
+            if (!QFile::rename(oldConfig.filePath(), newConfigFile)) {
+                qCWarning(lcApplication) << "Failed to rename configuration file from" << oldConfigFile << "to" << newConfigFile;
+            }
+        }
+    }
```

Comment:
Similar to above -- I would rename `oldConfigFile` to `oldConfigFileName` and `newConfigFile` to `newConfigFileName` to clarify what we are using here

Reply:
Same with `anyConfigFileList` to `anyConfigFileNameList`